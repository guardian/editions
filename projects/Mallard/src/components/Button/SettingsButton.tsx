import React from 'react'
import { Button, ButtonAppearance } from './Button'

const SettingsButton = ({ onPress }: { onPress: () => void }) => (
    <Button
        accessibilityLabel="Settings button"
        accessibilityHint="Navigates to the settings screen"
        accessibilityRole="button"
        icon={'\uE040'}
        alt="Settings"
        onPress={onPress}
        appearance={ButtonAppearance.skeleton}
    />
)

export { SettingsButton }
