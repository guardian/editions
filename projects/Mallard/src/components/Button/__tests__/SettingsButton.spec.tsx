import React from 'react';
import type { ReactTestRendererJSON } from 'react-test-renderer';
import TestRenderer from 'react-test-renderer';
import { SettingsButton } from '../SettingsButton';

describe('SettingsButton', () => {
	it('should display a SettingsButton', () => {
		const component: ReactTestRendererJSON | null = TestRenderer.create(
			<SettingsButton onPress={() => {}} />,
		).toJSON();
		expect(component).toMatchSnapshot();
	});
});
