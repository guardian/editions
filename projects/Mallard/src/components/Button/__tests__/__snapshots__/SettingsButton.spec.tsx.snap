// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsButton should display a SettingsButton 1`] = `
<View
  accessibilityHint="Navigates to the settings screen"
  accessibilityLabel="Settings button"
  accessibilityRole="button"
  accessible={true}
  collapsable={false}
  focusable={true}
  hitSlop={
    Object {
      "bottom": 10,
      "left": 10,
      "right": 10,
      "top": 10,
    }
  }
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 999,
          "height": 42,
          "justifyContent": "center",
          "paddingHorizontal": 21,
        },
        Object {
          "backgroundColor": undefined,
          "borderColor": "#121212",
          "borderWidth": 1,
        },
        Object {
          "alignItems": "center",
          "aspectRatio": 1,
          "justifyContent": "center",
          "paddingHorizontal": 0,
          "width": 42,
        },
        undefined,
      ]
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "fontFamily": "GuardianIcons-Regular",
            "fontSize": 20,
            "lineHeight": 20,
          },
          Array [
            Object {
              "color": "#121212",
            },
            undefined,
          ],
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;
